libs/wingc/src/type_check.rs

```diff
@@ -4495,9 +4526,59 @@ impl<'a> TypeChecker<'a> {
 				}
 			}
 		}
+
+		// Check that if the class implements sim.IResource, then the
```

Comment:
I think this should be checked for all inflight methods, not just if the class implements `sim.IResource`...

Reply:
Added a note in the PR description -- I think we can address this as a follow up